### PR TITLE
changed banner styling for the timeline page

### PR DESCRIPTION
**Pull request by Manasa2850**
## What this PR does
< solves issue #3648  >

## Screenshots
Before:
< 
Before
 >

After:
< 
WhatsApp Image 2020-01-10 at 12 52 01 AM
>

### CODE REVIEW

File: app/assets/javascripts/components/timeline/timeline.jsx [hunk 1]
```diff
@@ -64,6 +64,13 @@ const Timeline = createReactClass({
     return week.blocks;
   },
 
+  submit(e) {
```
Comment (ragesoss) [inline app/assets/javascripts/components/timeline/timeline.jsx]: I don't understand what's going on with all the changes to this file. It doesn't look like this `submit` function is used anywhere.

Can you explain more fully what the strategy of this change is, and clean up any changes to this file that aren't necessary?

Comment (Manasa2850) [reply app/assets/javascripts/components/timeline/timeline.jsx]: @ragesoss sorry my bad. There's some unwanted code in the file. I'll remove that and make a commit

Comment (Manasa2850) [reply app/assets/javascripts/components/timeline/timeline.jsx]: And about the other changes, basically what I've done is I check whether there is a banner or not and do the styling accordingly. I got the if conditions from course_alerts.jsx

Comment (ragesoss) [reply app/assets/javascripts/components/timeline/timeline.jsx]: Thanks for the explanation! That helps me understand it.

Comment (Manasa2850) [reply app/assets/javascripts/components/timeline/timeline.jsx]: Is what I have done fine?

**Comment by Manasa2850** [comment]
@ragesoss I wasn't able to solve it just with CSS. 
So ok, I'll try creating a new state element and work with it!

**Comment by Manasa2850** [comment]
@ragesoss this is the best I could do. I tried this method but it didn't work 
1) tried using **'document.getElementsByclassName('notification').offsetHeight; '** but it returned undefined for the height

**Comment by ragesoss** [comment]
Thanks for exploring this @Manasa2850. I'm pretty sure that something simpler and more general is possible, so I'm not going to merge a fix that introduces so much code to work around one specific situation. It looks like the current state of it is duplicating the logic of to show that specific banner, but there could be other large banners being rendered from the Notifications component that cause the same problems.

The `getElementsByClassName` function will return an array of elements with that class name; `offsetHeight` is undefined for the array, but will work for an individual element. It looks like the notification you're testing against is actually not from the Notifications component but from the CourseAlerts component. I think the place to start would be something like `document.getElementsByClassName('course-alerts')[0].offsetHeight`

**Comment by ragesoss** [comment]
(Or add an ID to that CourseAlerts element, since there should only be one.)

**Comment by ragesoss** [comment]
I'm going to close this for now. Please feel free to reopen if you take it up again.

**Pull request closed without merging**